SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2727,80 +2722,58 @@ void CSolver::SetResidual_BGS(CGeometry *geometry, CConfig *config) {
 
 void CSolver::SetRotatingFrame_GCL(CGeometry *geometry, CConfig *config) {
 
-  unsigned short iDim, nDim = geometry->GetnDim(), iVar, nVar = GetnVar(), iMarker;
-  unsigned long iVertex, iEdge;
-  su2double ProjGridVel;
-  const su2double* Normal;
-
-  /*--- Loop interior edges ---*/
-
-  for (iEdge = 0; iEdge < geometry->GetnEdge(); iEdge++) {
+  /*--- Loop interior points ---*/
 
-    const unsigned long iPoint = geometry->edges->GetNode(iEdge,0);
-    const unsigned long jPoint = geometry->edges->GetNode(iEdge,1);
+  SU2_OMP_FOR_STAT(roundUpDiv(nPointDomain,2*omp_get_max_threads()))
+  for (auto iPoint = 0ul; iPoint < nPointDomain; ++iPoint) {
 
-    /*--- Solution at each edge point ---*/
-
-    su2double *Solution_i = base_nodes->GetSolution(iPoint);
-    su2double *Solution_j = base_nodes->GetSolution(jPoint);
-
-    for (iVar = 0; iVar < nVar; iVar++)
-      Solution[iVar] = 0.5* (Solution_i[iVar] + Solution_j[iVar]);
```

Comment:
This function (`CSolver::SetRotatingFrame_GCL`) was averaging the solution but then updating the iPoint / jPoint residuals with Solution_i / j instead, if anyone knows of a good reason for that let me know and I'll put it back, otherwise "it's fixed".

Reply:
Actually this apparent fix makes one of the turbomachinery cases fail, so I've put the apparent bug back.

Reply:
Can you make an issue for this please with a link to the spot in the code? I think this could be a bug too but need to look through it...